metrics_utility/automation_controller_billing/helpers.py

```diff
@@ -1,9 +1,122 @@
 import json
 
 from itertools import chain
+from typing import Any, Dict, Optional, Tuple
 
 import pandas as pd
 
+from django.db import connection
+from django.utils.dateparse import parse_datetime
+
+from metrics_utility.logger import logger
+
+
+def get_last_entries_from_db() -> Optional[str]:
+    """
+    Get AUTOMATION_ANALYTICS_LAST_ENTRIES directly from database.
+
+    Returns:
+        Optional[str]: JSON string from database, or None if not found or error occurs
+    """
+    try:
+        with connection.cursor() as cursor:
+            cursor.execute("""
+                SELECT value
+                FROM conf_setting
+                WHERE key = 'AUTOMATION_ANALYTICS_LAST_ENTRIES'
+                LIMIT 1
+            """)
+            result = cursor.fetchone()
+
+            if result and result[0]:
+                return json.loads(result[0])  # This is the JSON value
```

Comment:
Looking at `collector.py` and `collectors.py`, both of them do:

```python
last_entries = get_last_entries_from_db()
last_gathered_entries = json.loads(last_entries or '{}', object_hook=datetime_hook)
```

but the value returned here is `json.loads(result[0])` .. so we have a double JSON->object transformation, and it will fail the second time.

But ..  only the other 2 `json.loads` apply that `object_hook=datetime_hook`, which seems to be necessary for comparing dates in last entries.

So... 2 options here:

```suggestion
                return result[0]  # Returning JSON now
```

which will make the other 2 places get the value they expect.

OR

```suggestion
                return json.loads(result[0] or '{}', object_hook=datetime_hook)
```

and then those other places can change to just...

```diff
-        last_entries = get_last_entries_from_db()
-        last_gathered_entries = json.loads(last_entries or '{}', object_hook=datetime_hook)
+        last_gathered_entries = get_last_entries_from_db()
```